### PR TITLE
Fix missing order numbers from transaction report CSV

### LINKED ISSUE #7592: Transactions Report - Fix missing order numbers in CSV
### Describe the bug

The order numbers linked to some transactions seem randomly missing in the exported CSV even though they are visible on the Transactions list page.

### To Reproduce

1. Go to WP Admin → Payments → Transactions.
2. Click the "Download" button on the top-right corner of the transactions list table.
3. Download the CSV received in the email.

### Actual behavior

Order numbers are missing for some transactions.

### Expected behavior

Order numbers for all transactions should be available where applicable.

### Screenshots

Order nos. missing in CSV:

Screenshot 2023-11-01 at 1 44 33 PM

But visible on the Transactions page:

image

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #7592 

#### Changes proposed in this Pull Request
For WooPayments Subscriptions that uses [Stripe Billing](https://stripe.com/en-gb-ro/billing), update transaction to contain also the order id.

#### Testing instructions

**Enable Stripe Billing subscriptions**

1. In `wp_options` table, set `_wcpay_feature_subscriptions` to `1`
2. Setup WCPay Dev tools with Test Clocks setup. FG article available here: PCYsg-DQ4-p2 

We need to test it together with 4209-gh-Automattic/woocommerce-payments-server

**On develop branch:**
1. Purchase a subscription product to create a Stripe Billing subscription
2. In the admin panel, go to Woo->Subscription and go to the new created subscription. In the left you will find Dev tools panel, with the test clock
Screenshot 2023-11-22 at 17 17 43

3. enable testing mode, in order to trigger the next payment events.
Screenshot 2023-11-22 at 17 30 10

4. trigger `invoice.paid` event
5. Go to `Payments->Transactions` and download the report. The order id is missing from the report
6. Switch to this branch and test again. The order id should be added to the report for the new transactions.
7. A demo can be found http://d.pr/v/R0zak0 . 

*

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7769` or branch name `fix/7592-update-transaction-order-id` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7592-update-transaction-order-id) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 6cece53ad65dc63abb516632309b856c5fe0d029
- Build time: 2023-12-06 13:08:14 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/subscriptions/class-wc-payments-invoice-service.php [hunk 1]
```diff
@@ -314,16 +314,41 @@ public function get_and_attach_intent_info_to_order( $order, $intent_id ) {
...
+	 * Update a charge with the order id from invoice.
+	 *
+	 * @param array $invoice
+	 * @param int   $order_id
+	 *
+	 * @return void
+	 * @throws API_Exception
+	 */
+	public function update_charge_details( array $invoice, int $order_id ) {
+
```
Comment (mordeth) [inline includes/subscriptions/class-wc-payments-invoice-service.php]: I believe the empty line is unnecessary.

**Review by mordeth — APPROVED** [review]
✅ The code appears to be in good shape! (Added a minor nitpick; feel free to ignore it.)
✅ Done manual testing with subscription renewals, and the order number is displayed correctly.
Screenshot 2023-11-28 at 13 22 11

File: includes/subscriptions/class-wc-payments-invoice-service.php (hunk 1 above)
Comment (dpaun1985) [reply includes/subscriptions/class-wc-payments-invoice-service.php]: fixed in https://github.com/Automattic/woocommerce-payments/pull/7769/commits/fb986fca8cbec7b3e2536095dab550ca416553b5. thx

**Review by mattallan — APPROVED** [review]
This approach looks good to me.
I've tested this locally and confirmed the renewal order IDs for Stripe Billing subscriptions are populated in the charges and transactions tables 💯

**Pull request merged**